File: fingertip/src/main/java/com/praszapps/fingertip/model/di/module/AppModule.java
package com.praszapps.fingertip.model.di.module;

import android.content.Context;

import com.praszapps.fingertip.view.FingertipApplication;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module()
public class AppModule {
    @Provides
    @Singleton
    Context provideContext(FingertipApplication application) {
        return application.getApplicationContext();
    }

}

File: fingertip/src/main/java/com/praszapps/fingertip/model/di/module/AppModule.kt
package com.praszapps.fingertip.model.di.module

import android.content.Context
import com.praszapps.fingertip.view.FingertipApplication
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AppModule {
    @Provides
    @Singleton
    internal fun provideContext(application: FingertipApplication): Context {
        return application.applicationContext
    }

}